NitroxClient/GameLogic/Spawning/WorldEntities/WorldEntitySpawnerResolver.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections.Generic;
+using NitroxModel.DataStructures.GameLogic.Entities;
+using NitroxModel_Subnautica.DataStructures;
+
+namespace NitroxClient.GameLogic.Spawning.WorldEntities
+{
+    public class WorldEntitySpawnerResolver
+    {
+        private readonly DefaultWorldEntitySpawner defaultEntitySpawner = new DefaultWorldEntitySpawner();
+        private readonly CellRootSpawner cellRootSpawner = new CellRootSpawner();
+        
+        private readonly Dictionary<TechType, IWorldEntitySpawner> customSpawnersByTechType = new Dictionary<TechType, IWorldEntitySpawner>();
+
+        public WorldEntitySpawnerResolver()
+        {            
+            customSpawnersByTechType[TechType.Crash] = new CrashEntitySpawner();
+            customSpawnersByTechType[TechType.Reefback] = new ReefbackWorldEntitySpawner(defaultEntitySpawner);
+        }
+
+        public IWorldEntitySpawner ResolveEntitySpawner(WorldEntity entity)
+        {
+            if (entity.ClassId == "55d7ab35-de97-4d95-af6c-ac8d03bb54ca")
```

Comment:
Is there a way to detect properly that it is a CellRoot spawner ?
Or it is the only way

Reply:
What do you mean with properly? The classId is a unique identifier which ties the entity to a particular prefab.

Reply:
I might be a unique ID, but it still hardcoded in our code

Reply:
Yes it's hardcoded but will probably never change. It survived the SN update after all.

Reply:
I do agree with dartasen that it might be problematic should this value change. I assume there is no way for us to get this programmically from subnautica itself?

Reply:
There is probably but just with another magic string which can change.

Reply:
Just a thought: this might be a good way to utilize the new entity system. We can check for top-level items that have no parent an initialize them as a CellRootEntity (which doesn't have all the fancy features of WorldEntity).  Then in CellEntities packet was can send a List<CellRootEntity> which have their WorldEntity as children.  

However, since this code is pre-existing, and simply moved from one place to another, I will explore such a refactor as a follow-up.